keep storage class as cold for restored objects

welansari opened the pull request:
To be compliant with the AWS S3 standard, the storage class of restored objects should be left as cold location

Issue: CLDSRV-400

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/cloudserver/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-400 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.7.24`


Please check the `Fix Version/s` of CLDSRV-400, or the target
branch of this pull request.

francoisferrand reviewed (commented):
- i get this is missing some tests (in objectHead for exemple), to check he storage class is as expected when the object is 'restored
- there seems to be some extra processing of x-amz-storage-class for multipart : so we should ensure this change does not affect the behavior when restoring a multipart object (e.g. when cold backend would need to do a multipart put to restore)
- `x-amz-storage-class` seems to be used a bit everywhere: so i fear that in non-versionned (or version-suspended), the 'cold' storage class would remain set when the object is replaced... → should be tested as well?

Commit:
keep storage class as cold for restored objects

To be compliant with the AWS S3 standard, the storage class
of restored objects should be left as cold location

Issue: CLDSRV-400

Commit:
test keeping same storage class when restoring a cold object

Issue: CLDSRV-400

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-400 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.7.25`


Please check the `Fix Version/s` of CLDSRV-400, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-400 contains:


* `8.7.25`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.7.25`

* `8.8.0`


Please check the `Fix Version/s` of CLDSRV-400, or the target
branch of this pull request.

nicolas2bert commented:
We use the `storageclass` to prevent an object to be lifecycled twice.
Updating `x-amz-storage-class` seems to not impact this logic, however, I would suggest you to double-check that `storageclass` metadata value is not assigned to `x-amz-storage-class` or vice versa somewhere in the code.

welansari commented:
@nicolas2bert i've looked into backbeat and cloudserver, the only instances where i see the values of `storageclass` and `x-amz-storage-class`  getting assigned together is in the case of a lifecycle transition and expiration of a restored object.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-400 contains:


* `8.7.26`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.7.26`

* `8.8.1`


Please check the `Fix Version/s` of CLDSRV-400, or the target
branch of this pull request.

welansari commented:
/approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/CLDSRV-400` into
`development/8.7`
* [w/8.8/improvement/CLDSRV-400](https://github.com/scality/cloudserver/compare/development/8.8...w/8.8/improvement/CLDSRV-400) will be merged into `development/8.8` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.7`

* :heavy_check_mark: `development/8.8`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.7`

* :heavy_check_mark: `development/8.8`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`



Please check the status of the associated issue CLDSRV-400.

Goodbye kerkesni.
